### PR TITLE
ord: Print a simpler message when write_db fails to open file.

### PR DESCRIPTION
**Pull request by titan73**

### CODE REVIEW

**Comment by titan73** [comment]
For https://github.com/The-OpenROAD-Project/OpenROAD/issues/6120.

File: src/OpenRoad.cc [hunk 1]
```diff
@@ -507,9 +507,12 @@ void OpenRoad::writeDb(std::ostream& stream)
...
 void OpenRoad::writeDb(const char* filename)
 {
-  utl::StreamHandler stream_handler(filename, true);
-
-  db_->write(stream_handler.getStream());
+  try {
+    utl::StreamHandler stream_handler(filename, true);
+    db_->write(stream_handler.getStream());
+  } catch (const std::ios_base::failure& f) {
+    logger_->error(ORD, 55, "Cannot write database to {}", filename);
```
Comment (maliberty) [inline src/OpenRoad.cc:514]: Please include the f.what string in the message as it may give more detail on the cause of the failure.

Comment (titan73) [reply src/OpenRoad.cc:514]: Yes for write_lef. write_def is different with the '.1'. Where does it come from?
```
>>> write_def foo/bar
[ERROR GUI-0070] No such file or directory: foo/bar.1
>>> write_lef foo/bar
[ERROR GUI-0070] basic_ios::clear: iostream error (failed to open 'foo/bar.1'): iostream error
```

Comment (maliberty) [reply src/OpenRoad.cc:514]: write_lef might need to write multiple files so they get .<num> suffixes.

write_lef/def are not part of GUI.

Comment (titan73) [reply src/OpenRoad.cc:514]: What is the case where write_lef write multiple files?
Regarding write_db & write_def, why using a suffix then? Should we use basic iostream instead?

**Review by oharboe — CHANGES_REQUESTED** [review]
If we are going to do this, then the util::StreamHandler would be the place to do it.

As I recall util::StreamHandler went through a number of iterations and lots of review feedback, a lots of tripwires...

**Comment by titan73** [comment]
> If we are going to do this, then the util::StreamHandler would be the place to do it.
> 

I don't understand your suggestion.

> As I recall util::StreamHandler went through a number of iterations and lots of review feedback, a lots of tripwires...

What is the purpose of util::StreamHandler? What problem is it supposed to solve? OR uses std::ifstream for reading.

**Comment by oharboe** [comment]
@titan73 master for the code you propose to change is using utl::StreamHandler today. I think the  code reads pretty well and the purpose is pretty clear from the code.

If you ask ChatGPT/copilot to explain what it does and why it exists instead of your proposed change, I am pretty sure you will get a satisfactory answer.

In this day and age, I consider simple code that ChatGPT/copilot can adequately explain purpose and implementation of sufficiently documented.

**Comment by titan73** [comment]
No need for ChapGPT. Seems the code prevents the existing file to be overwritten in case of failure.
I close this PR for now. I'll come up with something else later.

**Pull request closed without merging**